listener/src/main.rs

```diff
@@ -0,0 +1,232 @@
+extern crate env_logger;
+extern crate graph;
+extern crate graph_core;
+extern crate graph_datasource_ethereum;
+extern crate lazy_static;
+
+mod runtime_host;
+mod link_resolver;
+
+use lazy_static::lazy_static;
+use std::env;
+use std::sync::Arc;
+use std::time::Duration;
+
+use graph::components::forward;
+use graph::prelude::*;
+use graph::log::logger;
+use graph::tokio_executor;
+use graph::tokio_timer;
+use graph::tokio_timer::timer::Timer;
+
+use graph_core::{SubgraphInstanceManager, SubgraphRegistrar, SubgraphAssignmentProvider};
+
+use graph_datasource_ethereum::{BlockStreamBuilder, Transport};
+
+use graph_server_http::GraphQLServer as GraphQLQueryServer;
+use graph_server_websocket::SubscriptionServer as GraphQLSubscriptionServer;
+
+use graph_store_postgres::{Store as DieselStore, StoreConfig};
+
+use runtime_host::RustRuntimeHost;
+use link_resolver::LocalLinkResolver;
+
+lazy_static! {
+    static ref ANCESTOR_COUNT: u64 = 50;
+    static ref REORG_THRESHOLD: u64 = 50;
+    static ref NODE_ID: NodeId = NodeId::new("default").unwrap();
+    static ref BLOCK_POLLING_INTERVAL: Duration = Duration::from_millis(1000);
+}
+
+fn main() {
+    use std::sync::Mutex;
+    use tokio::runtime;
+
+    // Create components for tokio context: multi-threaded runtime, executor
+    // context on the runtime, and Timer handle.
+    //
+    // Configure the runtime to shutdown after a panic.
+    let runtime: Arc<Mutex<Option<runtime::Runtime>>> = Arc::new(Mutex::new(None));
+    let handler_runtime = runtime.clone();
+    *runtime.lock().unwrap() = Some(
+        runtime::Builder::new()
+            .core_threads(100)
+            .panic_handler(move |_| {
+                let runtime = handler_runtime.clone();
+                std::thread::spawn(move || {
+                    if let Some(runtime) = runtime.lock().unwrap().take() {
+                        // Try to cleanly shutdown the runtime, but
+                        // unconditionally exit after a while.
+                        std::thread::spawn(|| {
+                            std::thread::sleep(Duration::from_millis(3000));
+                            std::process::exit(1);
+                        });
+                        runtime
+                            .shutdown_now()
+                            .wait()
+                            .expect("Failed to shutdown Tokio Runtime");
+                        println!("Runtime cleaned up and shutdown successfully");
```

Comment:
`println!` - could we make this a log instead!

Reply:
```suggestion
                        info!("Runtime cleaned up and shutdown successfully");
```

Reply:
There is no logger available in this scope. I hope having this one off print statement is ok? We could try to instantiate it outside async_main and pass it in but I'm not sure if it's that important. Again, most of this setup glue code is taken from the graph node.

Reply:
Good enough... just wanted to point it out. If I recall correctly, the python linter would fail if there were any dangling print statements.